### PR TITLE
Spark: Don't distribute records in DELETE if mode is none

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR makes DELETE respect the distribution mode set in table properties.

### CODE REVIEW

**Comment by aokolnychyi** [comment]
cc @dilipbiswal @rdblue

File: spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java [hunk 1]
```diff
@@ -320,7 +320,7 @@ public static Distribution buildRequiredDistribution(org.apache.iceberg.Table ta
     }
   }
 
-  private static DistributionMode getDistributionMode(org.apache.iceberg.Table table) {
+  public static DistributionMode getDistributionMode(org.apache.iceberg.Table table) {
```
Comment (rdblue) [inline spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java:323]: Minor: If this is going to be public, we should fix the name. `get` doesn't add any value. How about `distributionModeFor` instead?

**Review by rdblue — APPROVED** [review]

File: spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java (hunk 1 above)
Comment (aokolnychyi) [reply spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java:323]: I don't like adding `get` either but the method is doing something. I think `distributionModeFor` may be a bit better, I'll change.

Comment (aokolnychyi) [reply spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java]: Fixed.

**Comment by aokolnychyi** [comment]
Thanks for reviewing, @rdblue!

**Pull request merged**